shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2511,6 +2549,23 @@ - (void)takeKeyboardScreenshotAndDisplay {
   _keyboardViewContainer.frame = _keyboardRect;
 }
 
+- (BOOL)showEditMenu:(NSDictionary*)args API_AVAILABLE(ios(16.0)) {
+  if (!self.activeView.isFirstResponder) {
```

Comment:
@justinmc this is where I determine the menu is triggered by text input widgets vs non-text input widgets. 

I can also check `_textInputClient != 0`, but I have a feeling that there could be an edge case where `_textInputClient` is set, but the it is not yet in editing state. So I feel checking the first responder is safer.